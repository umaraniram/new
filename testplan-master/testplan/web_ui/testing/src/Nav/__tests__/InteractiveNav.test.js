/* Unit tests for the InteractiveNav component. */
import React from 'react';
import {shallow} from 'enzyme';
import {StyleSheetTestUtils} from "aphrodite";

import InteractiveNav from '../InteractiveNav.js';
import {FakeInteractiveReport} from '../../Common/sampleReports.js';

describe('InteractiveNav', () => {
  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
  });

  afterEach(() => {
    // Resume style injection once test is finished.
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('shallow renders and matches snapshot', () => {
    const renderedNav = shallow(
      <InteractiveNav
        report={FakeInteractiveReport}
        selected={[FakeInteractiveReport]}
        filter={null}
        displayEmpty={true}
        displayTags={false}
        displayTime={false}
        handleNavClick={jest.fn()}
        handlePlayClick={jest.fn()}
        envCtrlCallback={jest.fn()}
      />
    );
    expect(renderedNav).toMatchSnapshot();
  });
});

