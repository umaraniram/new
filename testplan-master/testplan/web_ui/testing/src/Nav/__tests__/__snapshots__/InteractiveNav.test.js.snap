// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InteractiveNav shallow renders and matches snapshot 1`] = `
<Fragment>
  <NavBreadcrumbs
    entries={
      Array [
        Object {
          "counter": Object {
            "failed": 0,
            "passed": 0,
          },
          "entries": Array [
            Object {
              "category": "multitest",
              "counter": Object {
                "failed": 0,
                "passed": 0,
              },
              "description": null,
              "entries": Array [
                Object {
                  "category": "testsuite",
                  "counter": Object {
                    "failed": 0,
                    "passed": 0,
                  },
                  "description": null,
                  "entries": Array [
                    Object {
                      "counter": Object {
                        "failed": 0,
                        "passed": 0,
                      },
                      "description": null,
                      "entries": Array [],
                      "logs": Array [],
                      "name": "test_interactive",
                      "name_type_index": Array [],
                      "runtime_status": "ready",
                      "status": "unknown",
                      "status_override": null,
                      "suite_related": false,
                      "tags": Object {},
                      "tags_index": Object {},
                      "timer": Object {},
                      "type": "TestCaseReport",
                      "uid": "ddd",
                    },
                  ],
                  "logs": Array [],
                  "name": "Interactive Suite",
                  "name_type_index": Array [],
                  "part": null,
                  "runtime_status": "ready",
                  "status": "unknown",
                  "status_override": null,
                  "tags": Object {},
                  "tags_index": Object {},
                  "timer": Object {},
                  "type": "TestGroupReport",
                  "uid": "ccc",
                },
              ],
              "logs": Array [],
              "name": "Interactive MTest",
              "name_type_index": Array [],
              "part": null,
              "runtime_status": "ready",
              "status": "unknown",
              "status_override": null,
              "tags": Object {},
              "tags_index": Object {},
              "timer": Object {},
              "type": "TestGroupReport",
              "uid": "bbb",
            },
          ],
          "meta": Object {},
          "name": "Fake Interactive Report",
          "name_type_index": Array [],
          "runtime_status": "ready",
          "status": "unknown",
          "status_override": null,
          "tags_index": Object {},
          "timer": null,
          "uid": "aaa",
        },
      ]
    }
    handleNavClick={[MockFunction]}
  />
  <InteractiveNavList
    autoSelect={[Function]}
    breadcrumbLength={1}
    displayEmpty={true}
    displayTags={false}
    displayTime={false}
    entries={
      Array [
        Object {
          "category": "multitest",
          "counter": Object {
            "failed": 0,
            "passed": 0,
          },
          "description": null,
          "entries": Array [
            Object {
              "category": "testsuite",
              "counter": Object {
                "failed": 0,
                "passed": 0,
              },
              "description": null,
              "entries": Array [
                Object {
                  "counter": Object {
                    "failed": 0,
                    "passed": 0,
                  },
                  "description": null,
                  "entries": Array [],
                  "logs": Array [],
                  "name": "test_interactive",
                  "name_type_index": Array [],
                  "runtime_status": "ready",
                  "status": "unknown",
                  "status_override": null,
                  "suite_related": false,
                  "tags": Object {},
                  "tags_index": Object {},
                  "timer": Object {},
                  "type": "TestCaseReport",
                  "uid": "ddd",
                },
              ],
              "logs": Array [],
              "name": "Interactive Suite",
              "name_type_index": Array [],
              "part": null,
              "runtime_status": "ready",
              "status": "unknown",
              "status_override": null,
              "tags": Object {},
              "tags_index": Object {},
              "timer": Object {},
              "type": "TestGroupReport",
              "uid": "ccc",
            },
          ],
          "logs": Array [],
          "name": "Interactive MTest",
          "name_type_index": Array [],
          "part": null,
          "runtime_status": "ready",
          "status": "unknown",
          "status_override": null,
          "tags": Object {},
          "tags_index": Object {},
          "timer": Object {},
          "type": "TestGroupReport",
          "uid": "bbb",
        },
      ]
    }
    envCtrlCallback={[MockFunction]}
    filter={null}
    handleNavClick={[MockFunction]}
    handlePlayClick={[MockFunction]}
    selectedUid="aaa"
  />
</Fragment>
`;
